### ISSUE TITLE
Enable Che to more easily support latest runtime containers images through automation

### ISSUE THREAD

**Issue by l0rd** [eclipse-che/che#15799]
### Is your enhancement related to a problem? Please describe.

Problem is that we have an important amount of stacks (devfiles that reference container images and code samples) and those should be updated pretty often. A recent example is the issue related to the Gradle stack that hasn't been updated to gradle 6 https://github.com/eclipse/che/issues/15723 (november 2019).

### Describe the solution you'd like

- [ ] Leveraging tools like [dependabot](https://dependabot.com/) to be notified (or even auto submit PRs) when a new versions of an application that we install in a stack is released.
- [ ] Setup some automated tests for every single stack that verifies that the stack doesn't get broken

**Comment by tsmaeder** [comment]
Hmh...while dependabot works for docker files it doesn't really do anything for VS Code extensions or for images we mention in devfiles or plugin meta.ymls

**Comment by ericwill** [comment]
> Hmh...while dependabot works for docker files it doesn't really do anything for VS Code extensions or for images we mention in devfiles or plugin meta.ymls

We have a similar task ahead of us on the plugins side, and it seems that automation of the VS Code extensions will need to be done via script/GitHub actions.

**Comment by rhopp** [comment]
Is this somethin QE team should work on, or will this be handled by languages team?

**Comment by tsmaeder** [comment]
I don't think this is a core QE issue: devfiles usually have an author that is familiar with the kinks of a particular technology. It might be argued that this is a task for the plugins team, since it is really not specific to languages (asciidoctor, anyone?)

**Comment by tsmaeder** [comment]
The problem is that usually we need to update plugins in tandem stacks, for example we need tooling able to handle Java 13 if we want to have a devfile to program for that platform. So just updating to "latest" is many times not the right thing to do. 
I think the value of any solution here will be in the detection of changes in the underlying plugins and images: creating a PR is usually not the time-consuming task.

**Comment by ericwill** [comment]
Even though the languages/plugins responsibilities are shared, the actual plugins for the languages side of things still live in the plugin registry, which falls on my team. And since we are working towards a common goal, we should probably agree on what this automation will/should look like.

I agree that the opening the PR's isn't the issue, it's being notified of new upstream changes and then handing off the work to the appropriate developer to do the update. My initial thought is to have a script that runs once a day, does a check for all the upstream repositories, and then opens issues in the che-plugin-registry (or che) for any plugins that need updating. From there the issues can be triaged according to team responsibility. WDYT?
